src/Chart.tsx

```diff
@@ -138,36 +134,26 @@ export const Chart = forwardRef<ChartHandle, ChartProps>(
 		const { controlledHoverSignal, selectedIdSignalName, selectedSeriesSignalName } = useSpecProps(spec);
 		const chartConfig = useMemo(() => getChartConfig(config, colorScheme), [config, colorScheme]);
 
-		// Need to de a deep copy of the data because vega tries to transform the data
-		const chartData = useMemo(() => {
-			const clonedData = JSON.parse(JSON.stringify(data));
-
-			// We received a full Vega data array with potentially multiple dataset objects
-			if (isVegaData(clonedData)) {
-				return extractValues(clonedData);
-			}
-
-			// We received a simple array of data and we'll set a default key of 'table' to reference internally
-			return { [TABLE]: clonedData };
-		}, [data]);
-
 		useEffect(() => {
 			const tooltipElement = document.getElementById('vg-tooltip-element');
 			if (!tooltipElement) return;
 			// Hide tooltips on all charts when a popover is open
 			tooltipElement.hidden = popoverIsOpen;
 
 			// if the popover is closed, reset the selected data
-			if (!popoverIsOpen) {
+			if (!popoverIsOpen && chartView.current) {
 				selectedData.current = null;
+
+				setTimeout(() => {
+					chartView.current?.runAsync();
+				}, 1000);
 			}
-		}, [popoverIsOpen]);
+		}, [popoverIsOpen, chartView]);
```

Comment:
`chartView` is a `ref` object, so it will never trigger this `useEffect()`. If you want the `useEffect()` to run when `chartView.current` changes you have a few options:
1. Make `chartView` a stateful variable (probably not desired)
2. Remove `chartView` from the dependency list

There's also a third option which is to keep it in (as `chartView.current`), but be aware that updates to `chartView.current` _won't trigger a rerender_. But if `chartView.current` is different from one render to the next, it will trigger this `useEffect()`.

Reply:
This is old and I don't need this in there anymore.